app/lib/.server/llm/stream-text.ts

```diff
@@ -64,13 +63,20 @@ export function streamText(
       return { ...message, content };
     }
 
-    return message; // No changes for non-user messages
+    return message; 
   });
 
+  const modelDetails = MODEL_LIST.find((m) => m.name === currentModel);
+
+  const dynamicMaxTokens =
+    modelDetails && modelDetails.maxTokenAllowed
+      ? Math.min(MAX_TOKENS, modelDetails.maxTokenAllowed)
```

Comment:
So, I tested it and it works.
I am just wondering, why is this here?
Why are we not using just model tokens? Why min of both?

Reply:
Hi @wonderwhy-er,

Thanks for testing and sharing your feedback! Here's why I use `Math.min(MAX_TOKENS, modelDetails.maxTokenAllowed)`:

1. **Safety Check:** `MAX_TOKENS` acts as a safeguard to prevent issues if `modelDetails.maxTokenAllowed` is misconfigured or unexpectedly high.
2. **System Limits:** It ensures the token limit stays within the project's resource constraints, even if a model allows more tokens.
3. **Compatibility:** `MAX_TOKENS` was likely used in the code before, so keeping it avoids breaking any existing functionality.

If this adds unnecessary complexity, we can simplify the logic and rely only on `modelDetails.maxTokenAllowed`. Let me know what you think!

Reply:
I would use it as default if model does not have its own but allow it to be bigger than MAX_TOKENS

So if model does not have its own maxTokenAllowed then MAX_TOKENS is used.

You have code like this

  const dynamicMaxTokens =
    modelDetails && modelDetails.maxTokenAllowed
      ? Math.min(MAX_TOKENS, modelDetails.maxTokenAllowed)
      : MAX_TOKENS;
      
      Change it to
      
        const dynamicMaxTokens =
    modelDetails && modelDetails.maxTokenAllowed
      ? modelDetails.maxTokenAllowed
      : MAX_TOKENS;